scripts/metavar.py

```diff
@@ -1186,6 +1191,56 @@ def array_ref(self, local_name=None):
         match = FORTRAN_SCALAR_REF_RE.match(local_name)
         return match
 
+    def intrinsic_elements(self, check_dict=None):
+        """Return a list of the standard names of this Var object's 'leaf'
+        intrinsic elements or this Var object's standard name if it is an
+        intrinsic 'leaf' variable.
+        If this Var object cannot be reduced to one or more intrinsic 'leaf'
+        variables (e.g., a DDT Var with no named elements), return None.
+        A 'leaf' intrinsic Var is a Var of intrinsic Fortran type which has
+        no children. If a Var has children, those children will be searched
+        to find leaves. If a Var is a DDT, its named elements are searched.
+        If <check_dict> is not None, it is checked for children if none are
+        found in this variable (via finding a variable in <check_dict> with
+        the same standard name).
+        Currently, an array of DDTs is not processeed (return None) since
+        Fortran does not support a way to reference those elements.
```

Comment:
I am not sure of the implication of this, hope we can clarify at the ccpp-framework meeting.

Reply:
I guess it depends on whether or not the NOAA models or the SCM will ever need this feature. We generate code to make sure that all input variables in the active physics suite have been initialized. To do this, we need to be able to look into arrays of fields or into DDTs to find fields that could be initialized (2D or 3D physical fields). Is this something you will likely need?

Reply:
I am not sure. Maybe we should look at some specific examples of how the UFS works at the moment in our next developer meeting (or in a separate meeting) to see if this is already an issue or not.

Reply:
Okay. You can prepare by answering the question:
At run time, does / will the UFS need to query the standard names used by a suite?

Reply:
I don't think so. We have never used the standard names at run time (after switching to the static build 2 years ago), only at build time.